security/pkg/pki/ra/k8s_ra.go

```diff
@@ -52,15 +65,29 @@ func NewKubernetesRA(raOpts *IstioRAOptions) (*KubernetesRA, error) {
 		return nil, raerror.NewError(raerror.CAInitFail, fmt.Errorf("error processing Certificate Bundle for Kubernetes RA"))
 	}
 	istioRA := &KubernetesRA{
-		csrInterface:                 raOpts.K8sClient,
-		raOpts:                       raOpts,
-		keyCertBundle:                keyCertBundle,
+		csrInterface: raOpts.K8sClient,
+		raOpts:       raOpts,
+		// CertSignerDomain is based on CERT_SIGNER_DOMAIN env variable
 		certSignerDomain:             raOpts.CertSignerDomain,
 		caCertificatesFromMeshConfig: make(map[string]string),
 	}
+	// This must be set - even if empty.
+	istioRA.keyCertBundle = keyCertBundle
 	return istioRA, nil
 }
 
+// kubernetesSign will use the Kubernetes CSR API to sign - the call may include a 'certSigner' from the incoming
+// gRPC metadata 'CertSigner', allowing the use of different signers for different workloads.
+// This feature requires 'certSignerDomain' to be set -  based on CERT_SIGNER_DOMAIN env variable - which is used as
+// prefix. If CERT_SIGNER_DOMAIN is empty the only cert signer is set via K8S_SIGNER, which is also the default if
+// 'CertSigner' metadata is not set.
+//
+// For example, K8S_SIGNER=issuers.cert-manager.io/sandbox.my-issuer can be used as default, and
+// CERT_SIGNER_DOMAIN=issuers.cert-manager.io will allow users to pick different issuers.
+// Istio does not check the value of the issuer.
```

Comment:
How does the user pick different issuers and with what granularity? Is this behavior documented somewhere?

Reply:
Nothing is documented or reviewed AFAIK. Someone added a header with no docs or review, and users can send arbitrary strings that may or may not select unexpected signers and create security issues.